### PR TITLE
[Keyboard] Add s-ol/0xC.pad

### PR DESCRIPTION
**Pull request by s-ol**
## Description

This PR adds a new keyboard (`0xc_pad`) and manufacturer folder (`s_ol`).
After an "Interest Check" last year, the first set of PCBs, Cases and Keycaps are currently in production.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Previous stale PR #11622 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by s-ol** [comment]
A note on status: When the last PR went stale, I had applied the feedback received from @fauxpark and @noroadsleft, except for [this suggestion](https://github.com/qmk/qmk_firmware/pull/11622#discussion_r588540689) which seems to be in conflict with the keymap template outlined in documentation.

File: keyboards/s_ol/0xc_pad/info.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "keyboard_name": "0xC.pad",
+    "url": "https://s-ol.nu/0xC.pad",
+    "debounce": 5,
+    "diode_direction": "ROW2COL",
+    "matrix_pins": {
+      "rows": ["B7", "B6", "B5", "B4"],
+      "cols": ["D4", "D3", "D2", "D1"]
+    },
+    "width": 4, 
+    "height": 4,
```
Comment (fauxpark) [inline keyboards/s_ol/0xc_pad/info.json:11]:
```suggestion
```

File: keyboards/s_ol/0xc_pad/rules.mk [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = atmega32u2
+LTO_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/s_ol/0xc_pad/rules.mk:3]:
```suggestion
```

File: keyboards/s_ol/0xc_pad/rules.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/s_ol/0xc_pad/rules.mk:23]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by s-ol** [comment]
@fauxpark I applied all your suggestions and squashed them into the original commit.

**Review by drashna — APPROVED** [review]

File: keyboards/s_ol/0xc_pad/0xc_pad.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ */
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT(                                                 \
+                                                                \
+    k01, k05, k02, k06, k10, k03, k07, k11, k15, k08, k12, k16) \
+                                                                \
+    { {k01, k05, KC_NO, KC_NO}, {k02, k06, k10, KC_NO}, {k03, k07, k11, k15}, {KC_NO, k08, k12, k16}, }
```
Comment (noroadsleft) [inline keyboards/s_ol/0xc_pad/0xc_pad.h:24]: Admittedly, this is a nitpick, but these are supposed to resemble the assembled board.

```suggestion
#define LAYOUT(         \
         k01, k05,      \
       k02, k06, k10,   \
    k03, k07, k11, k15, \
       k08, k12, k16    \
) { \
    { k01,   k05, KC_NO, KC_NO }, \
    { k02,   k06, k10,   KC_NO }, \
    { k03,   k07, k11,   k15   }, \
    { KC_NO, k08, k12,   k16   }  \
}
```

File: keyboards/s_ol/0xc_pad/readme.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+A 12-key, RGB-backlit hexagonal macropad.
+
+* Keyboard Maintainer: [s-ol](https://github.com/s-ol)
+* Hardware Supported: "hex12pad r0", 0xC.pad r1
+* Hardware Availability: [IC form](https://s-ol.nu/0xC.pad)
+
+To flash the example keymap for this macropad
+(after setting up your build environment):
+​
+    make s_ol/0xc_pad:default:flash
```
Comment (noroadsleft) [inline keyboards/s_ol/0xc_pad/readme.md:14]: You have a couple of zero-width spaces in this file that break the intended rendering.

```suggestion
(after setting up your build environment):

    make s_ol/0xc_pad:default:flash
```

File: keyboards/s_ol/0xc_pad/readme.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+Before flashing, the macropad must be brought into DFU mode by one of the
+following means:
+- holding down the top-left key (next to the USB port) while plugging it in
+- bridging the two pads labeled `BOOT` on the back of the PCB with a metallic
+  object (tweezers, paperclip. wire)
+- putting the `RESET` keycode in your keymap and pressing the corresponding key
+​
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/s_ol/0xc_pad/readme.md:23]:
```suggestion
- putting the `RESET` keycode in your keymap and pressing the corresponding key

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Additionally, you don't _need_ both `keyboards/s_ol/0xc_pad/keymaps/default/keymap.c` and `keyboards/s_ol/0xc_pad/keymaps/default/keymap.json`. I believe `keymap.json` takes precedence over `keymap.c` if one tries to compile when both are present, but I could be wrong about that.

File: keyboards/s_ol/0xc_pad/0xc_pad.h (hunk 4 above)
Comment (s-ol) [reply keyboards/s_ol/0xc_pad/0xc_pad.h]: Ah, well good you spotted this - it used to look like that but I must've run clang-format once too many.

File: keyboards/s_ol/0xc_pad/readme.md (hunk 5 above)
Comment (s-ol) [reply keyboards/s_ol/0xc_pad/readme.md]: Thanks, I'll investigate my editor settings to understand how this happened (and why I didn't notice)

**Comment by s-ol** [comment]
@noroadsleft it didn't seem like the `.json` was being used by `qmk compile`, but I originally added it because I thought this was how the QMK Configurator gets its default keymaps. I removed it now that I found the actual source directory in that repo.

**Comment by s-ol** [comment]
@fauxpark @noroadsleft If you have another moment, your change requests have all been applied.

**Comment by noroadsleft** [comment]
> @noroadsleft I followed the keymap documentation template found [here](https://docs.qmk.fm/#/documentation_templates), is that not up to date?

@s-ol, regarding [this suggestion to edit the keymap readme](https://github.com/qmk/qmk_firmware/pull/11622#discussion_r588540689), I don't know that it's *officially* outdated, but I find the readmes more useful - speaking personally rather than "professionally" - if text is given priority over images.

I'll note the sample on the Documentation Templates page was written in 2017 and hasn't fundamentally changed since then. Perhaps it *is* outdated... :thinking:

**Review by noroadsleft — APPROVED** [review]

File: keyboards/s_ol/0xc_pad/0xc_pad.c [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
```
Comment (fauxpark) [inline keyboards/s_ol/0xc_pad/0xc_pad.c:16]:
```suggestion
#include "0xc_pad.h"
```

File: keyboards/s_ol/0xc_pad/info.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+                {"x":0.5, "y":3},
+                {"x":1.5, "y":3},
+                {"x":2.5, "y":3}
+            ]
+        }
+    },
+    "usb": {
+      "vid": "0x732D",
+      "pid": "0x0C70",
+      "device_ver": "0x0001"
```
Comment (fauxpark) [inline keyboards/s_ol/0xc_pad/info.json:31]:
```suggestion
      "device_version": "0.0.1"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by s-ol** [comment]
Thanks @fauxpark, I applied those two changes and took the chance to rebase onto master as well.

File: keyboards/s_ol/0xc_pad/info.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+                {"x":0.5, "y":3},
+                {"x":1.5, "y":3},
+                {"x":2.5, "y":3}
+            ]
+        }
+    },
+    "usb": {
+      "vid": "0x732D",
+      "pid": "0x0C70",
+      "device_ver": "0.0.1"
```
Comment (fauxpark) [inline keyboards/s_ol/0xc_pad/info.json:31]:
```suggestion
      "device_version": "0.0.1"
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/s_ol/0xc_pad/info.json (hunk 9 above)
Comment (s-ol) [reply keyboards/s_ol/0xc_pad/info.json:31]: ah okay, I guess lib/python/qmk/info.py:77 should also be adjusted at some point then?

Comment (fauxpark) [reply keyboards/s_ol/0xc_pad/info.json]: https://github.com/qmk/qmk_firmware/pull/14580

Comment (s-ol) [reply keyboards/s_ol/0xc_pad/info.json]: I see 👍 Change is applied.

**Pull request merged**